packages/ra-auth-auth0/src/authProvider.ts

```diff
@@ -0,0 +1,204 @@
+ * import {
+ *    Admin,
+ *    Resource,
+ *    CustomRoutes,
+ *    AuthProvider,
+ *    DataProvider,
+ * } from 'react-admin';
+ * import { Route } from 'react-router-dom';
+ * import comments from './comments';
+ * import i18nProvider from './i18nProvider';
+ * import Layout from './Layout';
+ * import posts from './posts';
+ * import users from './users';
+ * import tags from './tags';
+ * import { Auth0AuthProvider, httpClient } from 'ra-auth-auth0';
+ * import { Auth0Client } from '@auth0/auth0-spa-js';
+ * import jsonServerProvider from 'ra-data-json-server';
+ *
+ * const getPermissions = (roles: String[]) => {
+ *    if (!roles) {
+ *        return false;
+ *    }
+ *    if (roles.includes('admin')) return 'admin';
+ *    if (roles.includes('user')) return 'user';
+ *    return false;
+ * };
+ *
+ *  const App = () => {
+ *    const [auth0, setAuth0] = useState(undefined);
+ *    const authProvider = useRef<AuthProvider>(undefined);
+ *    const dataProvider = useRef<DataProvider>(undefined);
+ *
+ *    useEffect(() => {
+ *        const initAuth0Client = async () => {
+ *           const clientAuth0 = new Auth0Client({
+ *              domain: import.meta.env.VITE_AUTH0_DOMAIN,
+ *              clientId: import.meta.env.VITE_AUTH0_CLIENT_ID,
+ *              cacheLocation: 'localstorage',
+ *              authorizationParams: {
+ *                  audience: import.meta.env.VITE_AUTH0_AUDIENCE,
+ *              },
+ *          });
+ *
+ *          authProvider.current = Auth0AuthProvider(clientAuth0, {
+ *              onPermissions: getPermissions,
+ *              loginRedirectUri: import.meta.env.VITE_LOGIN_REDIRECT_URL,
+ *              logoutRedirectUri: import.meta.env.VITE_LOGOUT_REDIRECT_URL,
+ *          });
+ *
+ *          const httpClientAuth0 = await httpClient(clientAuth0);
+ *          dataProvider.current = jsonServerProvider(
+ *              import.meta.env.VITE_API_URL,
+ *              httpClientAuth0
+ *          );
+ *
+ *          setAuth0(clientAuth0);
+ *       };
+ *       if (!auth0) {
+ *          initAuth0Client();
+ *       }
+ *    }, [auth0]);
+ *
+ *   if (!auth0) {
+ *     return <div>Loading...</div>;
+ *   }
+ *
+ *   return (
+ *        <Admin
+ *            authProvider={authProvider.current}
+ *            dataProvider={dataProvider.current}
+ *            i18nProvider={i18nProvider}
+ *            title="Example Admin"
+ *            layout={Layout}
+ *         >
+ *           {permissions => (
+ *               <>
+ *                   <Resource name="posts" {...posts} />
+ *                   <Resource name="comments" {...comments} />
+ *                   <Resource name="tags" {...tags} />
+ *                   {permissions === 'admin' ? (
+ *                       <Resource name="users" {...users} />
+ *                   ) : null}
+ *               </>
+ *           )}
+ *       </Admin>
+ *    );
+ * };
+ * export default App;
+ *
+ * ```
+ *
+ * @param client the Auth0 client *
```

Comment:
```suggestion
 * @param client the Auth0 client
```

Besides other possible params are missing from the JSDoc